crates/prover-services/sharp-service/src/lib.rs

```diff
@@ -77,6 +77,13 @@ impl SharpProverService {
         let fact_checker = FactChecker::new(sharp_cfg.rpc_node_url, sharp_cfg.verifier_address);
         Self::new(sharp_client, fact_checker)
     }
+
+    pub fn with_test_settings(settings: &impl SettingsProvider, port: u16) -> Self {
```

Comment:
```suggestion
    #[cfg(test)]
    pub fn with_test_settings(settings: &impl SettingsProvider, port: u16) -> Self {
```

Reply:
This is not working in sharp client crate as it is in different environment so we will have to continue without #[cfg(test)]